### PR TITLE
Bump versions to 4.0.1 and add Release notes.

### PR DESCRIPTION
**Pull request by praveenkumarct**
Release preparation.

### CODE REVIEW

File: docs/RELEASE_NOTES.md [hunk 1]
```diff
@@ -26,7 +26,17 @@
...
-### 4.0.0 - Mar 26, 2021
+### 4.0.1 - Mar 16, 2021
+[Commits](https://github.com/commercetools/commercetools-sync-java/compare/4.0.0...4.0.1) |
+[Javadoc](https://commercetools.github.io/commercetools-sync-java/v/4.0.1/) | 
+[Jar](https://bintray.com/commercetools/maven/commercetools-sync-java/4.0.1)
+
+- ✨ **Enhancements** (1)
+    -  To improve performance and to avoid 414 request-URI too large error, the services are using chunking on the
+    list of input(keys or sku's) to query only the keys which are not in cache already and cache the results.
+    Related to the issue [659](https://github.com/commercetools/commercetools-sync-java/issues/659).
```
Comment (ahmetoz) [inline docs/RELEASE_NOTES.md:37]: The issue is not related to 659. It's already resolved.

File: docs/RELEASE_NOTES.md [hunk 2]
```diff
@@ -26,7 +26,17 @@
...
 7. Add Migration guide section which specifies explicitly if there are breaking changes and how to tackle them.
 -->
-### 4.0.0 - Mar 26, 2021
+### 4.0.1 - Mar 16, 2021
+[Commits](https://github.com/commercetools/commercetools-sync-java/compare/4.0.0...4.0.1) |
+[Javadoc](https://commercetools.github.io/commercetools-sync-java/v/4.0.1/) | 
+[Jar](https://bintray.com/commercetools/maven/commercetools-sync-java/4.0.1)
+
+- ✨ **Enhancements** (1)
+    -  To improve performance and to avoid 414 request-URI too large error, the services are using chunking on the
```
Comment (ahmetoz) [inline docs/RELEASE_NOTES.md:35]: What is the performance improvement in this change ? Also caching and other parts are not changed 🤔

Comment (praveenkumarct) [reply docs/RELEASE_NOTES.md]: https://github.com/commercetools/commercetools-sync-java/pull/689/commits/a070cd87333a7a3e1da7801d846f1596dec6686b

File: docs/RELEASE_NOTES.md (hunk 1 above)
Comment (praveenkumarct) [reply docs/RELEASE_NOTES.md]: https://github.com/commercetools/commercetools-sync-java/pull/689/commits/a070cd87333a7a3e1da7801d846f1596dec6686b

**Review by ahmetoz — APPROVED** [review]

**Pull request merged**